uaclient/system.py

```diff
@@ -17,6 +17,123 @@
 # N.B. this relies on the version normalisation we perform in get_platform_info
 REGEX_OS_RELEASE_VERSION = r"(?P<release>\d+\.\d+) (LTS )?\((?P<series>\w+).*"
 
+RE_KERNEL_UNAME = (
+    r"^"
+    r"(?P<version>[\d]+[.-][\d]+[.-][\d]+)"
+    r"-"
+    r"(?P<abi>[\d]+)"
+    r"-"
+    r"(?P<flavor>[A-Za-z0-9_-]+)"
+    r"$"
+)
+RE_KERNEL_PROC_VERSION_SIGNATURE = (
+    r"^"
+    r"(?P<version>[\d]+[.-][\d]+[.-][\d]+)"
+    r"-"
+    r"(?P<abi>[\d]+)"
+    r"[.-]"
+    r"(?P<subrev>[\d]+)"
+    r"(~(?P<hwerev>[\d.]+))?"
+    r"-"
+    r"(?P<flavor>[A-Za-z0-9_-]+)"
```

Comment:
Just double checking, the kernel will always have the `abi` and `flavor` information on the version signature ?

Reply:
Yes I believe so. This regex was taken from here: https://github.com/canonical/livepatch-server/blob/a33d53646acb739de4f4bcd8c5e7cf3f9bd8ec6f/params/kernel.go